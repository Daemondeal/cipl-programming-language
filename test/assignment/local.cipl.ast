pass
(block
    (let a before)
    (expression (call print a))
    (expression (assign a after))
    (expression (call print a))
    (expression (call print (assign a arg)))
    (expression (call print a))
)

