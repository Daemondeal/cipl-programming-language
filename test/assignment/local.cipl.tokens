Token          Lexeme         Line 
PASS           pass           1    
LINE_END       \n             2    
INDENT                        2    
LET            let            2    
IDENTIFIER     a              2    
EQUAL          =              2    
STRING         "before"       2     before
LINE_END       \n             3    
IDENTIFIER     print          3    
LEFT_PAREN     (              3    
IDENTIFIER     a              3    
RIGHT_PAREN    )              3    
LINE_END       \n             4    
IDENTIFIER     a              5    
EQUAL          =              5    
STRING         "after"        5     after
LINE_END       \n             6    
IDENTIFIER     print          6    
LEFT_PAREN     (              6    
IDENTIFIER     a              6    
RIGHT_PAREN    )              6    
LINE_END       \n             7    
IDENTIFIER     print          8    
LEFT_PAREN     (              8    
IDENTIFIER     a              8    
EQUAL          =              8    
STRING         "arg"          8     arg
RIGHT_PAREN    )              8    
LINE_END       \n             9    
IDENTIFIER     print          9    
LEFT_PAREN     (              9    
IDENTIFIER     a              9    
RIGHT_PAREN    )              9    
LINE_END       \n             9    
DEDENT                        9    
EOF                           9    
