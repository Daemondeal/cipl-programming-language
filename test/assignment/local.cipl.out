::stdout::
before
after
arg
arg

::stderr::
